src/vs/platform/tasks/common/taskLogService.ts

```diff
@@ -0,0 +1,56 @@
+/*---------------------------------------------------------------------------------------------
```

Comment:
I wondered about sharing this (since it's just the terminal's copied) and calling it `WorkspaceTaskService`. I'll try that now.

Reply:
We'll probably end up merging this into a base implementation with the terminal one. Let's wait to hear @sandy081's feedback on that.